java/src/main/java/com/xxAMIDOxx/xxSTACKSxx/menu/handlers/CreateCategoryHandler.java

```diff
@@ -19,28 +19,27 @@
 @Component
 public class CreateCategoryHandler extends MenuBaseCommandHandler<CreateCategoryCommand> {
 
-  private UUID categoryId;
-
   public CreateCategoryHandler(
       MenuRepository menuRepository, ApplicationEventPublisher applicationEventPublisher) {
     super(menuRepository, applicationEventPublisher);
   }
 
   @Override
   Optional<UUID> handleCommand(Menu menu, CreateCategoryCommand command) {
-    menu.setCategories(addCategory(menu, command));
+    UUID categoryId = UUID.randomUUID();
+    menu.setCategories(addCategory(menu, command, categoryId));
     menuRepository.save(menu);
+    command.setCategoryId(categoryId);
     return Optional.of(categoryId);
   }
 
   @Override
   List<MenuEvent> raiseApplicationEvents(Menu menu, CreateCategoryCommand command) {
     return Arrays.asList(
-        new MenuUpdatedEvent(command), new CategoryCreatedEvent(command, categoryId));
+        new MenuUpdatedEvent(command), new CategoryCreatedEvent(command, command.getCategoryId()));
   }
 
-  List<Category> addCategory(Menu menu, CreateCategoryCommand command) {
-    categoryId = UUID.randomUUID();
+  List<Category> addCategory(Menu menu, CreateCategoryCommand command, UUID categoryId) {
```

Comment:
can remove this category variable from being passed in the function since this variable should be set in the command  by this point